crates/wallet/src/wallet/mod.rs

```diff
@@ -1452,12 +1452,11 @@ impl Wallet {
             Some(ref drain_recipient) => drain_recipient.clone(),
             None => {
                 let change_keychain = self.map_keychain(KeychainKind::Internal);
-                let ((index, spk), index_changeset) = self
+                let ((_index, spk), index_changeset) = self
                     .indexed_graph
                     .index
                     .next_unused_spk(change_keychain)
```

Comment:
I'm wondering whether it will make sense to introduce a `KeychainTxOutIndex::next_unused_spk_without_revealing`.

```rust
    pub fn next_unused_spk_without_revealing(&self, keychain: K) -> Option<Indexed<ScriptBuf>> {
        let next_unused = self
            .unused_keychain_spks(keychain.clone())
            .next()
            .map(|(i, spk)| (i, spk.to_owned()));
        let (next_index_to_reveal, _) = self.next_index(keychain.clone())?;
        next_unused.or_else(|| {
            self.unbounded_spk_iter(keychain)
                .expect("descriptor must exist")
                .nth(next_index_to_reveal as _)
        })
    }

```

This way, we can avoid potentially-revealing an spk here (which might not be used), and avoid recording the new revealed index in `self.stage`.

We can use a new variable `let mut drain_keychain_index = Option::<(KeychainKind, u32)>::None` to actually do a mutation later on (when we determine that we will have a change output in our tx).

Reply:
Thinking through this what I think you're advocating is something like "next unused or else peek next index". Would you then reveal the next index in the event that the change address is used in the tx?

Also if we don't reveal it here are we relying on `index_of_spk` returning a value in order to mark it used? (which should be the case as long as the lookahead > 0)

Reply:
> are we relying on index_of_spk returning a value in order to mark it used?

I think no because we'll just use `drain_keychain_index` instead of `index_of_spk`

Reply:
I pushed 9bd6c4703fecedf4237dff3091e515d3d000951a to show what that would look like, but I'm not certain this is truly better than revealing one if needed. If we reveal an address but don't use it, then future calls to `create_tx` shouldn't cause a runaway of revealed addresses

Reply:
alternatively I'm considering the approach in 9341041cfdadc329f7850cf028059db84d4eb658